src/main/kotlin/mate/academy/service/FigureSupplier.kt

```diff
@@ -0,0 +1,34 @@
+package mate.academy.service
+
+import mate.academy.model.Circle
+import mate.academy.model.Color
+import mate.academy.model.Figure
+import kotlin.random.Random
+import kotlin.reflect.KClass
+import kotlin.reflect.full.primaryConstructor
+
+class FigureSupplier {
+    private val figuresQuantity = Figure::class.sealedSubclasses.size
+    private val colorSupplier = ColorSupplier()
+
+    fun getRandomFigure(): Figure {
+        val randomFigureImplementation = Figure::class.sealedSubclasses[Random.nextInt(figuresQuantity)]
+        return makeRandomInstance(randomFigureImplementation)
+    }
+
+    fun getDefaultFigure() = Circle(DEFAULT_SIZE, colorSupplier.getDefaultColor())
+
+    @Suppress("SpreadOperator")
+    private fun makeRandomInstance(clazz: KClass<out Figure>): Figure {
```

Comment:
Interesting way, but here is would be enough simpler approach with YOUR_ENUM.values().random() and when block.
Note that you would need to rewrite constructors for figures

Reply:
I have to advocate my approach. It offers much more loose coupling than one you suggest. To add new figure you just have to add class with figure implementation with no need to rewrite other modules (acc to Open-Close Principle of OOP), and all the rest will be done automatically. Whereas approach with 'when' requires editing of FigureSupplier class every time new figure is added.  First I've made an implementation with when block, but then decided to make something more interesting, like framework-style approach.

Reply:
Also I can add that with my approach FigureSupplier shouldn't know anything about figures and their structure, except a) default figure b) figure parameter can be either Int or Color

Reply:
No problem) Let's discuss it)

Please note that my aim is not to dictate you that only one way is right for solution of this task.
So, consider this discussion as study process. And if approving this task blocks you from further studying, please let me know via Mate chat, we'll manage it.

First of all let back to task goal -> Get practice in OOP approach. 
Our discussion is in code optimizing field rather than in direct one.

You solution works and displays quite advanced knowledge, that's great. 
However, this approach has some drawbacks:
- less obvious logic
- using reflection could be 'thin ice' and source of unpredictable issues
- as a result more time consumption for modifying and supporting (equal stakeholder's money)

Also, please check attached screenshots. There is an issue which was receive via adding another Figure-type class.
![Screenshot 2024-05-16 143359](https://github.com/mate-academy/kt-oop-figures/assets/83113910/b62dfb1d-7ba1-4d99-90b9-fd4e6a533791)
![Screenshot 2024-05-16 143319](https://github.com/mate-academy/kt-oop-figures/assets/83113910/f7dbddf0-6b6f-4567-bf9f-26d3cd472927)

Reply:
Sure. Because it breaks convention that figure parameters can be only Int or Color. If goal is to break the code it can be achieved easily))

Reply:
Ok, no problem, I'll rewrite code with 'when' block. I just wanted to have fun and write something less conventional

Reply:
> Sure. Because it breaks convention that figure parameters can be only Int or Color. If goal is to break the code it can be achieved easily))

Sorry, but no) There weren't such limitation. Please check task requirements
![Screenshot 2024-05-17 160628](https://github.com/mate-academy/kt-oop-figures/assets/83113910/2b801b16-787d-40aa-9978-8b7959f90811)

Reply:
> Ok, no problem, I'll rewrite code with 'when' block. I just wanted to have fun and write something less conventional

Mischief definitely managed ) 
It was fun